projects/storefrontlib/cms-structure/services/cms-guards.service.ts

```diff
@@ -25,13 +27,36 @@ export class CmsGuardsService {
     protected unifiedInjector: UnifiedInjector
   ) {}
 
+  protected featureConfigService = inject(FeatureConfigService);
+  protected guardsComposer = inject(GuardsComposer);
+
+  /**
+   * Executes all guards for the given `componentTypes` and returns an Observable that:
+   * - emits `true` if all those guards pass (emit `true`)
+   * - emits `false` or `UrlTree` immediately if any those guards fails (returns `false` or `UrlTree`)
+   */
   cmsPageCanActivate(
     componentTypes: string[],
     route: CmsActivatedRouteSnapshot,
     state: RouterStateSnapshot
   ): Observable<boolean | UrlTree> {
     const guards = this.cmsComponentsService.getGuards(componentTypes);
 
+    if (
+      this.featureConfigService.isEnabled('cmsGuardsServiceUseGuardsComposer')
+    ) {
+      const guardsInstances: CanActivate[] = guards
+        .map((guardClass) =>
+          getLastValueSync(this.unifiedInjector.get<CanActivate>(guardClass))
+        )
+        .filter(
+          (guardInstance): guardInstance is CanActivate =>
```

Comment:
`nitpick:` Consider extracting this logic to a separate function for readability reasons (my subjective opinion)

Reply:
Done